### DISCUSSION TITLE
Vault auth broken in 17.0

### DISCUSSION THREAD

**Discussion by jpathy** [keycloak/keycloak#10836] [Archive]
Hi,
I was using `KC_VAULT_HASHICORP_AUTHENTICATION_APP_ROLE_ROLE_ID` and `KC_VAULT_HASHICORP_AUTHENTICATION_APP_ROLE_SECRET_ID`
env vars in 16.1, seems like they are changed and i can't find any documentation related to how to use vault hashicorp auth?

I am using build time: `--vault-url=http://127.0.0.1:8200 --vault=hashicorp`
with `vault-kv-paths=` in keycloak.conf
I get the following error when it uses vault secrets:
````
Mar 21 11:24:34 th kc.sh[663119]: 2022-03-21 11:24:34,033 ERROR [org.keycloak.services] (executor-thread-3) KC-SERVICES0029: Failed to send email: java.lang.UnsupportedOperationException:>
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultAuthManager.login(VaultAuthManager.java:165)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultAuthManager.vaultLogin(VaultAuthManager.java:145)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultAuthManager.login(VaultAuthManager.java:116)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultAuthManager.login(VaultAuthManager.java:95)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultAuthManager.getClientToken(VaultAuthManager.java:79)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultKvManager.readSecret(VaultKvManager.java:36)
Mar 21 11:24:34 th kc.sh[663119]:         at io.quarkus.vault.runtime.VaultKvManager_ClientProxy.readSecret(Unknown Source)
Mar 21 11:24:34 th kc.sh[663119]:         at org.keycloak.quarkus.runtime.vault.QuarkusVaultProvider.obtainSecretInternal(QuarkusVaultProvider.java:52)
````

**Comment by Czini** [comment]
@jpathy have you managed in the end to configure the vault authentication using environment variables? I have the same issue on my side. If I configure the authentication in the quarkus.properties file and mount that to the container works but that's not the way to go.

**Reply by jpathy** [reply]
Unfortunately, no, iirc it got removed in some code changes, i haven't checked recently to see what's the status here.
